source/val/Function.cpp

```diff
@@ -93,15 +99,35 @@ spv_result_t Function::RegisterLoopMerge(uint32_t merge_id,
                                          uint32_t continue_id) {
   RegisterBlock(merge_id, false);
   RegisterBlock(continue_id, false);
-  cfg_constructs_.emplace_back(get_current_block(), &blocks_.at(merge_id),
-                               &blocks_.at(continue_id));
+  BasicBlock& merge_block = blocks_.at(merge_id);
+  BasicBlock& continue_block = blocks_.at(continue_id);
+  assert(merge_block.is_type(kBlockTypeUndefined));
+
+  current_block_->set_type(kBlockTypeLoop);
+  merge_block.set_type(kBlockTypeMerge);
+  continue_block.set_type(kBlockTypeContinue);
+  assert(current_block_ &&
+         "RegisterLoopMerge must be called when called within a block");
+  cfg_constructs_.emplace_back(ConstructType::kLoop, get_current_block(),
+                               &merge_block);
+  Construct& loop_construct = cfg_constructs_.back();
+  cfg_constructs_.emplace_back(ConstructType::kContinue, &continue_block,
+                               nullptr);
```

Comment:
Now that the exit argument defaults to nullptr, you can drop this third argument.